src/main/java/Task.java

```diff
@@ -21,6 +21,30 @@ public Task(SecretaryW.TaskType type, String description, String time1) {
         this.time1 = time1;
     }
 
+    // Constructor for creating TODO Task from file data
+    public Task(SecretaryW.TaskType type, String description, boolean isDone) {
+        this.type = type;
+        this.description = description;
+        this.isDone = isDone;
+    }
+
+    // Constructor for creating DEADLINE Task from file data
+    public Task(SecretaryW.TaskType type, String description, String time1, boolean isDone) {
+        this.type = type;
+        this.description = description;
+        this.time1 = time1;
+        this.isDone = isDone;
+    }
+
+    // Constructor for creating EVENT Task from file data
+    public Task(SecretaryW.TaskType type, String description, String time1, String time2, boolean isDone) {
+        this.type = type;
+        this.description = description;
+        this.time1 = time1;
+        this.time2 = time2;
```

Comment:
instead of time1, time2, use more distinguishable name
maybe startTime, endTime?